File: JavaScript/1_two_sum/solution.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(numbers, target) {
    var map = {},
        i,
        length = numbers.length,
        value,
        neededValue,
        result = [];
        
    for(i = 0; i < length; i++) {
        value = numbers[i];
        neededValue = target - value;
        if (neededValue in map) {
            result.push(map[neededValue] + 1);
            result.push(i + 1);
            break;
        } else {
            map[value] = i;
        }
    }
    return result;
};

File: Cpp/1_two_sum/solution.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        vector<int> result;
        if (nums.size() == 0) return result;
        
        int tmp;
        map<int, int> m;
        for (int i = 0; i < nums.size(); i ++)
        {
            tmp = target - nums[i];
            if (m.find(tmp) != m.end())
            {
                result.push_back(m[tmp]);
                result.push_back(i);
                break;
            }
            else m[nums[i]] = i;
        }
        return result;
    }
};